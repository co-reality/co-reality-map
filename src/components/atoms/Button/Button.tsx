// @ts-nocheck
import React from "react";

// Typings
import { ButtonProps } from "./Button.types";

// Styles
import * as S from "./Button.styles";
import { Link } from "react-router-dom";

const AppButton: React.FC<ButtonProps> = ({
  customClass,
  loading,
  onClick,
  text,
  type,
  customStyle,
  children,
  gradient,
  isLink = false,
  linkTo,
}) => {
  if (loading)
    return (
      <div className="spinner-border">
        <span className="sr-only">Loading...</span>
      </div>
    );

  if (isLink) {
    return (
      <S.Button
        as={Link}
        className={customClass}
        style={customStyle}
        // hasGradient={gradient}
        to={linkTo!}
      >
        {children ?? text}
      </S.Button>
    );
  }

  return (
    <S.Button
      className={customClass}
      style={customStyle}
      type={type}
      onClick={onClick}
      hasGradient={gradient}
    >
      {children ?? text}
    </S.Button>
  );
};

AppButton.defaultProps = {
  type: "button",
};

export default AppButton;
