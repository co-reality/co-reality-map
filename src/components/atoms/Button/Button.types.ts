import { CSSProperties } from "react";

type ButtonAsLink = {
  isLink: true;
  linkTo: string;
};

type NotLink = {
  isLink?: false;
  linkTo?: never;
};

type LinkProps = ButtonAsLink | NotLink;

export type ButtonProps = LinkProps & {
  customClass?: string;
  customStyle?: CSSProperties;
  gradient?: boolean;
  loading?: boolean;
  onClick?: () => void;
  text?: string;
  type?: "button" | "reset" | "submit";
};
