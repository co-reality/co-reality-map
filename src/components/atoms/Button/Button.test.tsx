import React from "react";
import renderer from "react-test-renderer";

// Test component
import Button from "./Button";

describe("<Button />", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<Button text="Hello World" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
