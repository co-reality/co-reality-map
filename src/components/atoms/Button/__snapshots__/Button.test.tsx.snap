// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="sc-bdfBwQ eqEdkr"
  type="button"
>
  Hello World
</button>
`;
